### PR TITLE
Add better hashCode() function for single int field objects.

### PR DESCRIPTION
**Pull request by jasperpotts**
Work on issue #35 to add advanced hash code function to generated code.

### CODE REVIEW

File: pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/ModelGenerator.java [hunk 1]
```diff
@@ -255,6 +255,37 @@ public void generate(final Protobuf3Parser.MessageDefContext msgDef,
 			).replaceAll("\n","\n"+FIELD_INDENT);
 		}
 
+		// Add here hashCode() for object with a single int field.
+		if (fields.size() == 1) {
+			FieldType fieldType = fields.get(0).type();
+			switch (fieldType) {
+				case ENUM, INT32, UINT32, SINT32, FIXED32, SFIXED32 -> {
```
Comment (jasperpotts) [inline pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/ModelGenerator.java:262]: I don't think we need this for enums. We do need to add it for 64bit integers as well though.

File: pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/ModelGenerator.java [hunk 2]
```diff
@@ -255,6 +255,37 @@ public void generate(final Protobuf3Parser.MessageDefContext msgDef,
...
+							"            x ^= x >>> 27;\n" +
+							"            x += x << 16;\n" +
+							"            x ^= x >>> 20;\n" +
+							"            x += x << 5;\n" +
+							"            x ^= x >>> 18;\n" +
+							"            x += x << 10;\n" +
+							"            x ^= x >>> 24;\n" +
+							"            x += x << 30;\n" +
+							"            return x;\n" +
+							"        }\n\n";
```
Comment (jasperpotts) [inline pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/ModelGenerator.java:281]: Think I prefer staying with multi-line block strings with variable substitution using replace() like we have everywhere else. Mainly just for constancy. There is still some old style with string formatting but we are almost all moved over to blocks with "$abc" style variables with replace()

Comment (LLITCHEV) [reply pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/ModelGenerator.java]: Addressed.

File: pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/ModelGenerator.java (hunk 1 above)
Comment (LLITCHEV) [reply pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/ModelGenerator.java]: Addressed.

**Review by LLITCHEV — APPROVED** [review]
Approving this per talk with Jasper. I need to do this since he created the PR.

**Pull request merged**